core/tests/puppeteer-acceptance-tests/blog-admin-tests/blog-editor-create-and-delete-draft-blog-post.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Accpetance Test for a Blog Post Editor to create and
+ * delete draft blog posts.
+ */
+
+const createNewUser = require(
+  '../puppeteer-testing-utilities/initializeUsers.js');
+
+const ROLE_BLOG_ADMIN = 'blog admin';
+const ROLE_BLOG_POST_EDITOR = 'blog post editor';
+
+let blogEditorCreateDraftAndDeleteDraftBlogPost = async function() {
+  const superAdmin = await createNewUser.superAdmin(
```

Comment:
The superadmin need to be made static or generic. Because many tests are going to require superadmin. But the functions of superadmin are very limited like giving roles, revoking roles, expecting roles to be that role etc. So maybe it can be made into a generic class and then passed to each test and initialized at beginnning of the test

Reply:
Yes @seanlip, but most test will require roles to be granted for that particular user's CUJs. For eg. Blog post User journey will require the role of either Blog admin or Blog editor to be granted to the user. Only then they are able to do any significant task. And yes there will be times when superadmin will not be required in any manner. There we can just ommit importing superadmin. What i am proposing is making a seperate generic class for superadmin, which as per our need ( when we want superadmin or not ), we can import to our test and intiliaze it at the beginning of test. What are your thoughts ?

Reply:
No, the point I'm making is that, as part of the process of "getting a new blog editor user", you have a temporary superadmin in that function being called who helps with setting that user up. But that superadmin doesn't feature at all in the main test body. They're just in the "create new user" function.